### PR TITLE
Add Alter Sit to SemCICSImpl

### PR DESCRIPTION
**Pull request by lewisja1406**
Please review alter sit code for Galasa conversion

### CODE REVIEW

**Comment by lewisja1406** [comment]
@hobbit1983 Please review changes

File: galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/build.gradle [hunk 1]
```diff
@@ -4,7 +4,7 @@ plugins {
 
 description = 'Galasa Internal SEM Manager'
 
-version = '0.21.0'
+version = '0.25.0'
```
Comment (hobbit1983) [inline galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/build.gradle:7]: Is there the equivalent of a release.yaml to incorporate this into the internal bom?

File: galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/src/main/java/dev/galasa/sem/internal/SemCicsImpl.java [hunk 2]
```diff
@@ -78,42 +82,75 @@ public SemCicsImpl(SemManagerImpl semManager, ICicstsManagerSpi cicstsManager, S
...
+			// Append the runtime CICS JCL to existing JCL
+			sb.append(runtimeJcl);
+
+			// Store the JCL ready for runtime
+			this.jcl = sb.toString();
+			
+		} catch (Exception e) {
+			
+            throw new SemManagerException("Unable to generate CICS region JCL", e);
+		}
```
Comment (hobbit1983) [inline galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/src/main/java/dev/galasa/sem/internal/SemCicsImpl.java:151]: can we isolate this try catch to just the line of code that could throw an exception if possible

File: galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/src/main/java/dev/galasa/sem/internal/SemCicsImpl.java [hunk 3]
```diff
@@ -457,7 +494,44 @@ public IZosBatchJob getRegionJob() throws CicstsManagerException {
...
+        if (job.getStatus() == JobStatus.ACTIVE) {
+        	throw new CicstsManagerException("The CICS region " + theCics.getApplid() + " with tag " + theCics.getTag() + " is still running. Cannot alter a running CICS region");
+        }
+        
+        for (CICSRegion cicsRegions : complex.getCICS()) {
+        	
+        	// Find the correct region
+        	if (cicsRegions.getApplid().equals(theCics.getApplid())) {
+        		
+        		Sit sitOverride = new Sit(sitParam, sitValue, "Overriden during runtime");
```
Comment (hobbit1983) [inline galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/src/main/java/dev/galasa/sem/internal/SemCicsImpl.java:516]: I 'think' there are two 'd' in Overridden - I know picky picky

File: galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/src/main/java/dev/galasa/sem/internal/SemCicsImpl.java [hunk 4]
```diff
@@ -78,42 +82,75 @@ public SemCicsImpl(SemManagerImpl semManager, ICicstsManagerSpi cicstsManager, S
...
+            
+        } catch (Exception e) {
+        	
+            throw new SemManagerException("Could not generate the CICS runtime JCL", e);
+        }    
     }
 
+    /**
+     * From a batch job, generate some JCL to startup CICS
+     * and activate the APPID of the CICS job. Store the JCL
```
Comment (hobbit1983) [inline galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/src/main/java/dev/galasa/sem/internal/SemCicsImpl.java:115]: APPLID

File: galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/src/main/java/dev/galasa/sem/internal/SemCicsImpl.java [hunk 5]
```diff
@@ -78,42 +82,75 @@ public SemCicsImpl(SemManagerImpl semManager, ICicstsManagerSpi cicstsManager, S
...
+     * 
+     * @param job
+     * @throws SemManagerException
+     */
+	private void buildCicsJcl(Job job) throws SemManagerException {
+
+		try {
+			
+			// Build some skeleton JCL
+			String runtimeJcl = semZosHandler.generateJCL(job);
```
Comment (hobbit1983) [inline galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/src/main/java/dev/galasa/sem/internal/SemCicsImpl.java:126]: it's not skeleton in the same way we use the term in artefact manager - can you just have a quick look at what this method does and amend the comment please

**Review by hobbit1983 — CHANGES_REQUESTED** [review]
Nothing major - just some comment clean up and spellings - Think I got most of the reviewing done when we buddied through this yesterday.  Nice to see alterSIT making it's way into Galasa!

File: galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/build.gradle (hunk 1 above)
Comment (hobbit1983) [reply galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/build.gradle:7]: I see you have updated

**Review by hobbit1983 — APPROVED** [review]

File: galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/src/main/java/dev/galasa/sem/internal/SemCicsImpl.java [hunk 6]
```diff
@@ -457,7 +493,90 @@ public IZosBatchJob getRegionJob() throws CicstsManagerException {
...
+    @Override
+    public void removeSit(@NotNull String sitParam) throws CicstsManagerException {
+    	
+    	// Get the CICS
+    	ICicsRegion theCics = cicstsManager.locateCicsRegion(this.getTag());
+    	
+    	// Check the restricted parameters
+    	if (sitParam.toUpperCase().equals("APPLID") | sitParam.toUpperCase().equals("GRPLIST") || sitParam.toUpperCase().equals("SYSID")) {
+    		throw new CicstsManagerException(sitParam.toUpperCase() + " is a restricted SIT parameter. It cannot be removed");
+       	}
```
Comment (hobbit1983) [inline galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/src/main/java/dev/galasa/sem/internal/SemCicsImpl.java:541]: There is a concern here that if we wanted to test what happened if you tried to start a CICS without these parameters we might have a problem, but I think we are OK

**Review by hobbit1983 — APPROVED** [review]

**Pull request merged**